### ISSUE TITLE
option blkdev is privileged (non-root users unable to mount sdcard)

### ISSUE THREAD

**Issue by kaihendry** [libfuse/libfuse#147]
Hi there, Archlinux user here with this [fuse.conf](https://s.natalian.org/2017-03-12/fuse.conf
). As reported in https://bbs.archlinux.org/viewtopic.php?id=215945 everytime I try mount my sdcard via 'FUSE exfat 1.2.6' I hit this **fusermount: option blkdev is privileged**. I'm using the latest https://www.archlinux.org/packages/extra/x86_64/fuse-common/ package.

I gather from https://github.com/libfuse/libfuse/blob/master/doc/mount.fuse.8#L36 that **allow_other** or **allow_root** should work, but they don't.

```
fusermount: unknown parameter in /etc/fuse.conf at line 10: 'allow_root'
fusermount: unknown parameter in /etc/fuse.conf at line 10: 'allow_other'
```

**Comment by Nikratio** [comment]
Could you confirm that this is really what happens? Note that fusermount complains about a parameter called `allow_other` in `/etc/fuse.conf`. However, the `fuse.conf` that you linked above does not contain such a parameter.

`allow_other` is not something that belongs in fuse.conf, but is typically passed to the filesystem on the command line. The option that *enables* the use of `allow_other` (and `allow_root`) is `user_allow_other` and that is what should be in `fuse.conf`.

**Comment by Nikratio** [comment]
Regarding `blkdev` being privileged: this is a design choice and has nothing to do with `allow_other` etc. I'd like to give you the rationale, but I'm drawing a blank at the moment (this was established long before I became maintainer. It may be possible to relax this now, but I wouldn't do so without first figuring out why the restriction was added back then).

**Comment by trapexit** [comment]
Speaking of blkdev... Is there much in the way of documentation for using
that feature?

**Comment by Nikratio** [comment]
Only what `grep blkdev` finds in `doc/` and `include/*.h`.

**Comment by trapexit** [comment]
Perhaps I'll experiment with it and offer a patch to include some documentation. 

It's odd that there are so many features almost completely undocumented.

**Comment by Nikratio** [comment]
Lacking further information about the original problem, I'm closing this issue for now. Please feel free to reopen with additional information.

**Comment by tbull** [comment]
One year later, next guy with the same problem. I'm trying to switch from Debian to Arch Linux, which gives me many headaches, among them this issue. Currently I try to mount my ExFAT formatted USB memory stick as a user.

After activating `user_allow_other` in fuse.conf, i get the same **fusermount: option blkdev is privileged**. (On Debian, the `user_allow_other` was not needed, worked nevertheless, no idea why.) I mount simply with `mount /dev/sdb1`, thereby referring to the line in `/etc/fstab`:

    /dev/sdb1    /mnt/usb0    exfat,vfat,msdos,fat,auto    rw,user,noauto,noexec,uid=roland,gid=roland,fmask=117,dmask=007    0  0

Fuse is supplied by these Arch packages installed on my system:
https://www.archlinux.org/packages/extra/x86_64/fuse2/
https://www.archlinux.org/packages/extra/x86_64/fuse-common/
There's also another package available, which I have not installed. (That wasn't a conscious choice.)
https://www.archlinux.org/packages/extra/x86_64/fuse3/
ExFAT is supplied by this package:
https://www.archlinux.org/packages/community/x86_64/exfat-utils/

That's the point, where I am now. For your entertainment, I append my previous adventures below. ;-)

Before that, I had to put myself into the `disk` group, because `/dev/sd*` block devices are owned by root:disk, and permission to access the block device was denied. (That wasn't necessary on Debian, either.) (The disk group has number 994 on Arch, which looks to me as if the group is not supposed to contain any users.)

Then, I got the error **fusermount: user has no write access to mountpoint /mnt/usb0**. So I handed the mount point over to group `disk` and `chmod g+w`. This makes me wonder, too, because as to my knowledge, the mount point permissions should be of no concern for the mount operation.
